### PR TITLE
Fix linting requirement build numbers

### PR DESCRIPTION
**Pull request by isuruf**
Checklist
* [x] Added a ``news`` entry

And also allow more than one spacing.

### CODE REVIEW

File: tests/test_lint_recipe.py [hunk 1]
```diff
@@ -827,6 +827,7 @@ def test_single_space_pins(self):
                 ],
                 "host": [
                     "python >= 2",
+                    "libcblas 3.8.* *netlib",
```
Comment (isuruf) [inline tests/test_lint_recipe.py]: @scopatz, this was failing in master

File: conda_smithy/lint_recipe.py [hunk 2]
```diff
@@ -401,11 +401,11 @@ def lintify(meta, recipe_dir=None, conda_forge=False):
             if "{{" in req:
                 continue
             parts = req.split()
-            if req.count(" ") > 1:
+            if len(parts) > 2 and parts[1] in ["!=", "=", "==", ">", "<", "<=", ">="]:
```
Comment (ocefpaf) [inline conda_smithy/lint_recipe.py]: Nice!

File: tests/test_lint_recipe.py (hunk 1 above)
Comment (scopatz) [reply tests/test_lint_recipe.py]: Ahh good catch

File: tests/test_lint_recipe.py [hunk 3]
```diff
@@ -838,17 +839,16 @@ def test_single_space_pins(self):
...
+            "``requirements: run: xonsh>1.0`` must contain a space between the "
             "name and the pin, i.e. ``xonsh >1.0``",
-            "``requirements: run: conda= 4.*`` must contain a single space between the "
+            "``requirements: run: conda= 4.*`` must contain a space between the "
             "name and the pin, i.e. ``conda =4.*``",
-            "``requirements: run: conda-smithy<=54.*`` must contain a single space "
+            "``requirements: run: conda-smithy<=54.*`` must contain a space "
             "between the name and the pin, i.e. ``conda-smithy <=54.*``",
         ]
+        print(filtered_lints)
```
Comment (scopatz) [inline tests/test_lint_recipe.py]: This probably isn't strictly needed, but it is in the tests

Comment (scopatz) [reply tests/test_lint_recipe.py]: so not a big deal

Comment (isuruf) [reply tests/test_lint_recipe.py]: Let me fix that.

**Pull request merged**